### ISSUE TITLE
Map card, new contrasted design: the "the dark_mode" option does not work

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/frontend#13333]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

On 2022.7 a new "more contrasted" design was provided for Map card.
The [`dark_mode`](https://www.home-assistant.io/dashboards/map/#dark_mode) option now works not as it did earlier.

Here is how it works now:

Light default theme:
```
type: vertical-stack
cards:
  - type: map
    entities:
      - person.ildar
  - type: map
    entities:
      - person.ildar
    dark_mode: false
```
![image](https://user-images.githubusercontent.com/71872483/182699819-6a20e216-6c02-4aa4-8a0a-3253bed00ee1.png)

```
    ...
    dark_mode: true
```
![image](https://user-images.githubusercontent.com/71872483/182699897-be66e06c-3634-46af-be1f-07852061b7d1.png)

Dark default theme:
```
    ...
    dark_mode: false
```
![image](https://user-images.githubusercontent.com/71872483/182700960-35d717d2-75a1-4b2c-be1d-68a1126da785.png)

```
    ...
    dark_mode: true
```
![image](https://user-images.githubusercontent.com/71872483/182700333-9027ebf7-8a2b-4221-bff0-263da460e280.png)

### Describe the behavior you expected

If the `dark_mode` is `true` then:
 - if theme is in a dark mode -> map is dark
 - otherwise -> map is light

If the `dark_mode` is `false` or not specified then:
 - map is light

### Steps to reproduce the issue

As described above.

### What version of Home Assistant Core has the issue?

2022.8.0

### What was the last working version of Home Assistant Core?

2022.7.7

### In which browser are you experiencing the issue with?

Chrome 103.0.5060.134

### Which operating system are you using to run this browser?

Win10x64

**Comment by Kertz1954** [comment]
Another explanation.

If the system theme is set to Dark as opposed to Auto or Light, the Map Card,  Dark Mode ? toggle switch has no effect on the map theme. It remains in Dark mode theme.

If the system theme is set to Light, then the Dark Mode ? toggle switch does change the map theme between Light or Dark.

![New Map Card](https://user-images.githubusercontent.com/67558244/182717627-7b744be8-8ded-41b3-b401-3c8b97c1bbe2.png)

**Comment by ildar170975** [comment]
> Another explanation.

Is it your explanation of THE CURRENT implementation or is it your proposed implementation?

If 1st:
then "to Dark as opposed to Auto or Light, the Map Card, Dark Mode ? toggle switch has no effect" is WRONG - seen above, the "default dark theme, dark_mode: false" case.

**Comment by Kertz1954** [comment]
This is the current method of operation which is incorrect.  Dark mode is selected as main theme.
Map Card as shown is showing Dark Mode without the toggle being set.

**Comment by ildar170975** [comment]
> Map Card as shown is showing Dark Mode without the toggle being set.

Note that in dark theme the un-toggled "dark_mode" makes the map more contrasted.

**Comment by Kertz1954** [comment]
Yes, but that is not the corrected function of the toggle switch. It's being mis-used.

**Comment by ildar170975** [comment]
I described the expected behaviour in the 1st post.

**Comment by Kertz1954** [comment]
I suggest you re-read your 1st post. It's not at all clear what you are trying to achieve from it.

**Comment by zsarnett** [comment]
The dark mode toggle only works if you are in light mode and want to have the map dark. No other functionality is offered.

**Comment by Kertz1954** [comment]
Then the toggle should be removed if you are in dark mode.

**Comment by ildar170975** [comment]
> The dark mode toggle only works if you are in light mode and want to have the map dark

This was the previous functionality which does not work now:
If the dark_mode is true then:
    if theme is in a dark mode -> map is dark
    otherwise -> map is light
If the dark_mode is false or not specified then:
    map is light

Isn't it?

**Comment by Kertz1954** [comment]
As a programmer with 48 years of experience, the toggle should not have been changed. The whole purpose of this update was to first change the Light theme to a more colorful  presentation which as been acheived by updating the Leaflet OpenStreetMap. The Leaflet Dark theme still presented a problem as it is really dark, but a workaround was implemented to make it more readable. 
A user should still have the option to decide which theme they wish to see in the map, regardless of the  base OS theme.

**Comment by zsarnett** [comment]
This is a Feature Request not an issue.

**Comment by ildar170975** [comment]
What exactly is a FR?

1. See a difference between 2022.8 & 2022.7.4:
old:
![image](https://user-images.githubusercontent.com/71872483/182764071-33a96bea-cd7e-4e2d-a3ad-d3d9414c16e2.png)
new:
![image](https://user-images.githubusercontent.com/71872483/182764095-85211562-f04e-4a6a-8294-70bcba33265e.png)

The "dark theme / dark_mode=false" is DIFFERENT.
Should it be?

2. According to the [docs](https://www.home-assistant.io/dashboards/map/#dark_mode):
![image](https://user-images.githubusercontent.com/71872483/182764459-fc70951f-09a7-4d36-b311-165e780afcef.png)
But in both the "2022.7.4" & "2022.8" versions "dark_mode: default" is not the same as "dark_mode: false".

3. As for my [thoughts "how it was working long time ago"](https://github.com/home-assistant/frontend/commits/dev/src/components/map/ha-map.ts):
I may be wrong. Do not have an old HA version to clarify how it was working long time ago.

But even when comparing 2022.7.4 & 2022.8, what I see is:
-- the "dark theme / dark_mode=false" case is different;
-- in both versions  "dark_mode: default" is not the same as "dark_mode: false" - which is specified in the docs.

**Comment by ildar170975** [comment]
up

**Comment by arduiacob** [comment]
it should be possible to choose a light theme for the map, when the main theme is dark

**Comment by mikulik86** [comment]
up

**Comment by ildar170975** [comment]
up

**Comment by ildar170975** [comment]
up

**Comment by 532910** [comment]
It's impossible to have white map with dark app interface. This is a bug, that should be fixed. Recent activity is not a reason to ignore it, please disable auto-closing.

**Comment by nickelliott** [comment]
It makes sense that, if a user doesn't define a dark_mode setting, the map card follows the theme setting, but if the user specifically specifies dark_mode: false it surely makes sense to respect that?  Why not do so?

**Comment by karwosts** [comment]
Fixed by #20541
